continuous-integration/check.sh

```diff
@@ -0,0 +1,248 @@
+    echo "     logging_handler = logging.StreamHandler()"                                                     >> what.patch
+
+    echo "diff --git a/jhbuild/utils/systeminstall.py b/jhbuild/utils/systeminstall.py"                       >> what.patch
+    echo "index 75b0849..d5d45f0 100644"                                                                      >> what.patch
+    echo "--- a/jhbuild/utils/systeminstall.py"                                                               >> what.patch
+    echo "+++ b/jhbuild/utils/systeminstall.py"                                                               >> what.patch
+    echo "@@ -428,7 +428,7 @@ class AptSystemInstall(SystemInstall):"                                         >> what.patch
+    echo " "                                                                                                  >> what.patch
+    echo "     def _install_packages(self, native_packages):"                                                 >> what.patch
+    echo "         logging.info(_('Installing: %(pkgs)s') % {'pkgs': ' '.join(native_packages)})"             >> what.patch
+    echo "-        args = self._root_command_prefix_args + ['apt-get', 'install']"                            >> what.patch
+    echo "+        args = ['apt-get', '-y', 'install']"                                                       >> what.patch
+    echo "         args.extend(native_packages)"                                                              >> what.patch
+    echo "         subprocess.check_call(args)"                                                               >> what.patch
+
+    patch -p1 < what.patch
+    cd -
+}
+
+function do_Build_JHBuild(){
+    echo
+    echo '-- Building JHBuild --'
+
+    # Build JHBuild
+    cd jhbuild
+    git log --pretty=format:"%h %cd %s" -1
+    echo
+    ./autogen.sh
+    make -sj2
+    make install
+    PATH=$PATH:~/.local/bin
+    git reset --hard HEAD
+}
+
+function do_Build_Mozilla(){
+    echo
+    echo '-- Building Mozilla 31 --'
+
+    # Build Mozilla Stuff
+    if [[ -n "$SHELL" ]]; then
+        export SHELL=/bin/bash
+    fi
+    jhbuild build mozjs31
+}
+
+function do_Build_Package_Dependencies(){
+    echo
+    echo "-- Building Dependencies for $1 --"
+    jhbuild list $1
+
+    # Build package dependencies
+    if [[ $BASE == "ubuntu" ]]; then
+        jhbuild sysdeps --install $1
+    fi
+    jhbuild build $(jhbuild list $1 | sed '$d')
+}
+
+function do_Save_Files(){
+    echo
+    echo '-- Saving build files --'
+    mkdir -p /cwd/SAVED/$OS
+
+    cp -r ~/jhbuild /cwd/SAVED/$OS/jhbuild
+    cp -r ~/.local  /cwd/SAVED/$OS/.local
+    echo '-- Done --'
+}
+
+function do_Get_Files(){
+    echo
+    echo '-- Restoring build files --'
+
+    cp -r /cwd/SAVED/$OS/jhbuild ~/jhbuild
+    cp -r /cwd/SAVED/$OS/.local  ~/.local
+    echo '-- Done --'
+}
+
+function do_Clean_Image(){
```

Comment:
It was used, but it is not needed now (or it is no longer needed). I will remove it.